### PR TITLE
Added gtsam 4.1.1

### PR DESCRIPTION
**Pull request by martinvl**
Specify library name and version:  **gtsam/4.1.1**

Bumps GTSAM to 4.1.1.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@martinvl Please, don't forget to sign the CLA listed above. Also, you need to update the `config.yml` file, located in gtsam folder too with the new version.

File: recipes/gtsam/all/conanfile.py [hunk 1]
```diff
@@ -121,6 +121,10 @@ def validate(self):
         if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
             raise ConanInvalidConfiguration ("GTSAM requires MSVC >= 15")
 
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.version) >= '4.1' \
+                and self.options.shared:
+            raise ConanInvalidConfiguration("GTSAM does not support shared builds on MSVC")
```
Comment (prince-chrismc) [inline recipes/gtsam/all/conanfile.py:126]:
```suggestion
            raise ConanInvalidConfiguration("GTSAM does not support shared builds on MSVC. see https://github.com/borglab/gtsam/issues/1087")
```

**Review by prince-chrismc — COMMENTED** [review]
One tiny nit otherwise this is good

**Comment by martinvl** [comment]
Awesome, thanks for the added link!

File: recipes/gtsam/all/conanfile.py [hunk 2]
```diff
@@ -121,6 +121,10 @@ def validate(self):
         if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
             raise ConanInvalidConfiguration ("GTSAM requires MSVC >= 15")
 
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.version) >= '4.1' \
```
Comment (uilianries) [inline recipes/gtsam/all/conanfile.py:124]:
```suggestion
        if is_msvc(self) and tools.Version(self.version) >= '4.1' \
```
https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc

Don't forget to add the new import

Comment (martinvl) [reply recipes/gtsam/all/conanfile.py]: Very nice! Pushed a commit which replaces the manual checks with `is_msvc()` now.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**